### ISSUE TITLE
Realm/RLMArray.h file not found. Could not build Object-C module 'Realm'

### ISSUE THREAD

**Issue by karzler** [realm/realm-cocoa#3071]
My Xcode version is 7.2 and Swift's version is 2.1.1
I am struggling to build my project with Realm and RealmSwift framework. 
While building the project, these are errors I receive:

    'Realm/RLMArray.h' file not found.
    Could not build Objective-C module 'Realm'

Taking into problems I might run with Cocoapods, I installed Realm from source. Followed the process provided here: https://realm.io/docs/swift/latest/

I later tried with Cocoapods, that makes things even worse. I further applied Bridging Header, thinking that it might help me out: http://stackoverflow.com/questions/29905271/unable-to-use-both-swift-and-obj-c-libs-with-cocoapods

No success yet!

**Comment by JasonZengJ** [comment]
@karzler  I  ran into the same issue before, and solved by using CocoaPods 0.39.0，the CocoaPods 1.0.0 caused this problem.

**Comment by karzler** [comment]
@JasonZengJ I still find it difficult. Now it throws that it couldn't find the underlying module 'RealmSwift'

**Comment by karzler** [comment]
@JasonZengJ Now, even the other frameworks are not being imported. Strange.

**Comment by karzler** [comment]
@JasonZengJ can you mention the precise steps you took? That could probably help me in looking at where am i going wrong?

**Comment by JasonZengJ** [comment]
You can try following steps:

1. Create an empty swift project for a test
2. use CocoaPods 0.39.0 to install RealmSwift ( ensure your CocoaPods is 0.39.0.If your CocoaPods has been 1.0.0 before, you should use `gem uninstall CocoaPods` to uninstall 1.0.0.)
3. Compare the test project's build settings with your project's build settings, and find the difference.

**Comment by karzler** [comment]
I got it resolved. Need to install Realm plugin explicitly. That didn't happen automatically.

**Comment by jpsim** [comment]
@karzler what do you mean by "Realm plugin"? What did you need to explicitly install?

Realm only makes two things that could ostensibly be called "plugins": our [LLDB script](https://github.com/realm/realm-cocoa/blob/master/plugin/rlm_lldb.py) and our [Xcode plugin](https://github.com/realm/realm-cocoa/tree/master/plugin). Neither of those are required to install Realm via CocoaPods.

**Comment by karzler** [comment]
@jpsim I had to manually install Realm plugin for Xcode. It was only after this that I was able to resolve my issue.

**Comment by jpsim** [comment]
The Xcode plugin is entirely unrelated to the installation of the Realm framework. It's more likely that some other change you made (such as running `pod install` a second time to work around #2701) solved the problem.

In any case, since you've resolved this on your end, I'll close this ticket.

**Comment by winkelsdorf** [comment]
Ran today into the same issue (previously using 0.39 cocoapods, now --pre release). 

As suggested fixed by issuing ```pod install``` instead of ```pod update```. Not related to any Xcode plugin but probably best practice to use ```install``` after major changes to CocoaPods itself.

**Comment by nicopuri** [comment]
I have the same issue but using `Carthage 0.15`. Did you find the reason why it fails? Should it be an issue with the compiler in how it builds the framework?

**Comment by winkelsdorf** [comment]
@nicopuri No, unfortunately I didn't find the reason, but to be honest I had no time to investigate further. Looks like it's not related to CocoaPods/Carthage but probably to a change in Realm's header structure?

I also need to correct my statement from above: I was **not** able to fix and eventually reverted back to cocoapods 0.39 to get it running again.

**Comment by 2dvisio** [comment]
Any news on this issue? Facing same problems here...

**Comment by bumstyla** [comment]
Same here i updated to cocoapods 1.0 and this started happening.

**Comment by guidev** [comment]
same here

**Comment by jbegham** [comment]
+1, Cocoapods 1.0 and RealmSwift throws this error mentioned by @karzler

**Comment by guidev** [comment]
this worked for me https://github.com/realm/realm-cocoa/issues/3551

**Comment by huguesbr** [comment]
+1 same issue... using Xcode Version 7.3 (7D175):

Unable to make RealmSwift work if I'm using Cocoapods 1.0...

Using `pod _0.39.0_ install`, fix the issue but it's not really a solution..

```
platform :ios, '8.0'
use_frameworks!

target 'XXX' do
  pod 'RealmSwift', '~> 0.98'

  target 'XXXTests' do
    inherit! :search_paths
  end

  target 'XXXUITests' do
    inherit! :search_paths
  end
end

```

**Comment by marco318** [comment]
+1 with cocoapods 1.0.0 and Swift 2.2

**Comment by juanpdelat** [comment]
I was having the same issue with cocoapods 1.0.1 after updating RealmSwift from 0.99 to 1.0.0, it somehow got fixed by deleting all the files related to cocoapods, even the pod file, from the Xcode project  and the machine, then doing a fresh pod init, install AND update.

I first removed from Xcode the Pods project, the pods folder and the pods frameworks, I did this by hitting delete and then Move to Trash. Then in Finder I deleted the Pods folder, the Podfile and Podfile.lock files. Then pod init... install... update.

**Comment by jleach** [comment]
+1 same issue with Cocoapods 1.0.1.

**Comment by jpsim** [comment]
@jleach CocoaPods 1.0.1 should work. Maybe there's an Xcode/CocoaPods cache invalidation issue? Try these steps:

```
pod cache clean Realm
pod cache clean RealmSwift
pod deintegrate || rm -rf Pods
pod install --verbose
```

---

**Note to everyone:** I'm locking this conversation to encourage everyone to file a new issue, taking care to fill out the [issue template](https://github.com/realm/realm-cocoa/blob/master/.github/ISSUE_TEMPLATE.md), so that we actually have enough information to help you. Thanks for your understanding.
